### PR TITLE
refactor: duplicate adminform submissions to /api/v3 endpoint

### LINKED ISSUE #1511: [API] Duplicate admin get form submissions related endpoints to their new /api counterparts
This issue tracks the straight duplication of viewing admin form submissions endpoints to their `/api` rooted endpoints to follow a RESTful hierarchy. 

The old endpoint should not be deleted for backwards compatibility (at least for a few weeks after the new endpoint has been released).

The current to new mapping is shown below:

\# | Existing endpoints | New endpoints<br>(note that all routes are prefixed with /api/v3) | Description
-- | -- | --|--
1 | GET /:formId/adminform/submissions/count | GET /admin/forms/:formId/submissions/count | retrieve number of submissions
2 | GET /:formId/adminform/submissions/download | GET /admin/forms/:formId/submissions/download | download storage mode submissions as csv (+ attachments)

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
Migrates old endpoints located at `/:formId/adminform` to new endpoint located at `/api/v3/admin/forms` 

Closes #1511 

## Solution
1. shifts old endpoints rooted at `/adminform/submissions` to new api root 
2. combined them under a single `AdminFormsSubmissionsRouter`. 

**Improvements**:
1. Combined validator middleware together with their handlers. The reasoning for this is that certain handlers require parameters to be valid before the handler can proceed and hence, the handler should be exported atomically together with the middleware. 
2. For such handlers, comments have been added onto the base function to denote that they are exported **soley** for testing purposes 

## Tests

1. Ported over existing tests from `adminform.routes.spec.ts` for `/count` and `/download`

## Manual Tests 
1. Create a storage mode form with attachment fields. Publish the form, go to it and submit a few times with attachments.
    - Go to the admin panel and click on data. There should be `x responses to date`. Click on export and perform both actions. The files should be downloaded successfully. 
    - Change the date range to a date in the past without submissions. Verify that you are unable to export csv (both with/without attachments) 
    - Change the date change to a valid date range and try to export csv again (with and without). The files should be downloaded successfully

### CODE REVIEW

File: src/app/modules/form/admin-form/admin-form.controller.ts
Comment (karrui) [inline src/app/modules/form/admin-form/admin-form.controller.ts:322]: should be `min` instead of `greater` or there will be regression T_T, see #1634

File: src/app/routes/api/v3/admin/forms/admin-forms.feedback.routes.ts
Comment (karrui) [inline src/app/routes/api/v3/admin/forms/admin-forms.feedback.routes.ts:9]: whoops

File: src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts
Comment (karrui) [inline src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts:1]: nit: name this file admin-forms.submission**s**?

File: src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts
Comment (karrui) [inline src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts:22]: can use `get` by itself without the `route` function since we probably have no other HTTP methods on this endpoint

File: src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts
Comment (karrui) [inline src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts:42]: same here

File: src/public/modules/forms/services/submissions.client.factory.js
Comment (karrui) [inline src/public/modules/forms/services/submissions.client.factory.js:50]: nit: don't need to split up the constants, just call it `ADMIN_FORMS_BASE` or something and have the full URL there since there are no other versions

**Review by karrui — CHANGES_REQUESTED** [review]
fix the regression, and double check whether tests have also changed since you started on this PR; see 
#1620, #1604, #1584

File: src/app/modules/form/admin-form/admin-form.controller.ts
Comment (seaerchin) [reply src/app/modules/form/admin-form/admin-form.controller.ts:322]: oh ok HAHAHAH will do!

File: src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts
Comment (seaerchin) [reply src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts:1]: will do

File: src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts
Comment (seaerchin) [reply src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts:22]: i did this on purpose because i felt like the `get` section should solely reflect the middleware functions called and the `route` portion should reflect the http path. 

even though there's only a single handler here, i think this convention would make it easier on the reader because of the semantic meaning attached to `route` and `get`

File: src/public/modules/forms/services/submissions.client.factory.js
Comment (seaerchin) [reply src/public/modules/forms/services/submissions.client.factory.js:50]: will combine both into `ADMIN_FORMS_PREFIX`

**Comment by seaerchin** [comment]
@karrui fyi, #1620 doesn't directly impact this PR (because it's about `feedback.service` but this PR operates on the `route` level).
i think i pulled after #1604 got merged so i got lucky there HAHA
#1584 is regarding `/submissions/metadata` and doesn't directly impact this issue.

File: src/app/modules/form/admin-form/admin-form.controller.ts
Comment (seaerchin) [reply src/app/modules/form/admin-form/admin-form.controller.ts]: ported over tests from #1634 and also added a slightly modified version for `/submissions/download`

File: src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts
Comment (karrui) [reply src/app/routes/api/v3/admin/forms/admin-forms.submission.routes.ts]: sure, will defer to you.

File: src/app/modules/form/admin-form/admin-form.controller.ts
Comment (karrui) [inline src/app/modules/form/admin-form/admin-form.controller.ts:399]: nit: instead of casting, can declare the type explicitly

```ts
export const handleCountFormSubmissions: RequestHandler[] = [
  ...
]
```

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts
Comment (karrui) [inline src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:514]: same

File: src/app/routes/api/v3/admin/forms/admin-forms.routes.ts
Comment (karrui) [inline src/app/routes/api/v3/admin/forms/admin-forms.routes.ts:16]: nit: >< Submission**s**Router

File: src/app/routes/api/v3/admin/forms/admin-forms.submissions.routes.ts
Comment (karrui) [inline src/app/routes/api/v3/admin/forms/admin-forms.submissions.routes.ts:9]: nit: 
```diff
- Count the number of submissions for a public form
+ Count the number of submissions for a form
```

**Review by karrui — APPROVED** [review]
lgtm , fix nits if you want

File: src/app/modules/form/admin-form/admin-form.controller.ts
Comment (seaerchin) [reply src/app/modules/form/admin-form/admin-form.controller.ts:399]: good point, will do so

Comment (seaerchin) [reply src/app/modules/form/admin-form/admin-form.controller.ts:399]: as per our discussion on slack, because `RequestHandler` requires a `ParamsDictionary` as its first type, this becomes alot harder to do without casting because ParamsDictionary isn't generic. will just cast :/

**Pull request merged**
